json_journal/data_analysis.py

```diff
@@ -82,3 +83,49 @@ def total_profit(full_account_name_balances):
             pass
     total = (income * (-1)) - expenses
     return total
+
+
+def cumulative_profit(ledger):
+    '''
+    Calculate the cumulative profit.
+
+    Parameters
+    ----------
+    Ledger : dict
+        Loaded json file.
+
+    Returns
+    -------
+    Dates : list
+        List of dates.
+    Cumulative_profit : list
+        List of cumulative total of profit.
+    '''
+    sorted_ledger = sorted(ledger, key=lambda d: d['date'])
+    dates = []
+    profits = []
+    for transaction in sorted_ledger:
+        transaction_date = transaction['date']
+        entries = transaction['entries']
+        profit = 0
+        for entry in entries:
+            amount = entry['amount']
+            account = entry['account']
+            if (account.startswith("expense:") or account.startswith("income")):
+                profit -= amount
+            else:
+                pass
+
+        if transaction_date in dates:
+            new_profit = profit + profits[-1]
+            profits[-1] = new_profit
+        else:
+            dates.append(transaction_date)
+            profits.append(profit)
+
+    cumulative_total = 0
+    cumulative_profits = []
+    for value in profits:
+        cumulative_total += value
+        cumulative_profits.append(cumulative_total)
```

Comment:
I think this logic can be included in the above for-loop.

Reply:
I think this needs to stay where it is but am happy to discuss

Reply:
To discuss

Reply:
done